test/L2EthToken.spec.ts

```diff
@@ -0,0 +1,251 @@
+    });
+
+    it("not called by bootloader", async () => {
+      const amountToMint: BigNumber = ethers.utils.parseEther("10.0");
+      await expect(l2EthToken.connect(walletTo.address).mint(walletFrom.address, amountToMint)).to.be.rejectedWith(
+        "Callable only by the bootloader"
+      );
+    });
+  });
+
+  describe("transfer", () => {
+    it("transfer successfully", async () => {
+      await l2EthToken.connect(bootloaderAccount).mint(walletFrom.address, ethers.utils.parseEther("100.0"));
+
+      const senderBalandeBeforeTransfer: BigNumber = await l2EthToken.balanceOf(walletFrom.address);
+      const recipientBalanceBeforeTransfer: BigNumber = await l2EthToken.balanceOf(walletTo.address);
+
+      const amountToTransfer = ethers.utils.parseEther("10.0");
+
+      await expect(
+        l2EthToken.connect(bootloaderAccount).transferFromTo(walletFrom.address, walletTo.address, amountToTransfer)
+      )
+        .to.emit(l2EthToken, "Transfer")
+        .withArgs(walletFrom.address, walletTo.address, amountToTransfer);
+
+      const senderBalanceAfterTransfer: BigNumber = await l2EthToken.balanceOf(walletFrom.address);
+      const recipientBalanceAfterTransfer: BigNumber = await l2EthToken.balanceOf(walletTo.address);
+      expect(senderBalanceAfterTransfer).to.be.eq(senderBalandeBeforeTransfer.sub(amountToTransfer));
+      expect(recipientBalanceAfterTransfer).to.be.eq(recipientBalanceBeforeTransfer.add(amountToTransfer));
+    });
+
+    it("no tranfser due to insufficient balance", async () => {
+      await l2EthToken.connect(bootloaderAccount).mint(walletFrom.address, ethers.utils.parseEther("5.0"));
+      const amountToTransfer: BigNumber = ethers.utils.parseEther("100000000000000000.0");
+
+      await expect(
+        l2EthToken.connect(bootloaderAccount).transferFromTo(walletFrom.address, walletTo.address, amountToTransfer)
+      ).to.be.rejectedWith("Transfer amount exceeds balance");
+    });
+
+    it("no transfer - require special access", async () => {
+      const maliciousWallet: Wallet = ethers.Wallet.createRandom().connect(provider);
+      await l2EthToken.connect(bootloaderAccount).mint(maliciousWallet.address, ethers.utils.parseEther("20.0"));
+
+      const amountToTransfer: BigNumber = ethers.utils.parseEther("20.0");
+
+      await expect(
+        l2EthToken.connect(maliciousWallet).transferFromTo(maliciousWallet.address, walletTo.address, amountToTransfer)
+      ).to.be.rejectedWith("Only system contracts with special access can call this method");
+    });
+  });
+
+  describe("balanceOf", () => {
+    it("walletFrom address", async () => {
+      const amountToMint: BigNumber = ethers.utils.parseEther("10.0");
+
+      await l2EthToken.connect(bootloaderAccount).mint(walletFrom.address, amountToMint);
+      const balance = await l2EthToken.balanceOf(walletFrom.address);
+      expect(balance).to.equal(ethers.utils.parseEther("115.0"));
```

Comment:
I guess it's 115 because this address was used in the previous test cases. Can you use a random address here as well? It would be more clear and will not depend on the state.

Reply:
Yes, you're right, we've noticed it here
https://github.com/matter-labs/era-system-contracts/pull/88#discussion_r1437174212